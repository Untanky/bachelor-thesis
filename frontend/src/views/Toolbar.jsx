import React from 'react';
import { useSelector } from 'react-redux';

import LanguageSelectBox from '../containers/LanguageSelectBox';
import ButtonLink from '../components/ButtonLink';

import './Toolbar.scss';

const Toolbar = () => {
  return (
    <div className="toolbar">
      {port}
      <span className="right-align">
        <ButtonLink to="/post/create">
          Create post
        </ButtonLink>
        <LanguageSelectBox />
      </span>
    </div>
  )
};

export default Toolbar;
