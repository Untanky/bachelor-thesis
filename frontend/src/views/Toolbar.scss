.toolbar {
  box-sizing: border-box;
  width: 100%;
  display: flex;
  background-color: #f5f5f5;
  padding: 1em;
  border-radius: 2em;
  margin: 1em 0;
  box-shadow: 0 4px 4px 0 rgba(0, 0, 0, 0.2);

  .left-align {
    float: left
  }

  .right-align {
    margin-left: auto;
  }

  > * > * {
    margin: 0 0.25em;
  }
}