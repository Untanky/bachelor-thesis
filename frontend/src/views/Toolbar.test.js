import React from 'react';
import Toolbar from './Toolbar';
import renderer from 'react-test-renderer';
import { StaticRouter } from 'react-router-dom';

jest.mock('../containers/LanguageSelectBox.jsx', () => ({
  __esModule: true,
  default: () => { return <div>false</div>; },
}));

it('renders correctly', () => {
  const tree = renderer
    .create(
      <StaticRouter>
        <Toolbar />
      </StaticRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
