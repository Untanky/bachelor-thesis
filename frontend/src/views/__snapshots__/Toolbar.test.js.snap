// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="toolbar"
>
  <span
    className="right-align"
  >
    <a
      className="button"
      href="/post/create"
      onClick={[Function]}
    >
      Create post
    </a>
    <div>
      false
    </div>
  </span>
</div>
`;
